### PR TITLE
Add test/Project.toml

### PR DESCRIPTION
**Pull request by jishnub**
This lets us instantiate the test environment, and locally carry out tests involving specific packages.

### CODE REVIEW

**Comment by johnnychen94** [comment]
`test/Project.toml` [requires Julia 1.2+](https://pkgdocs.julialang.org/v1/creating-packages/#Test-specific-dependencies-in-Julia-1.2-and-above), we might just use [TestEnvs](https://github.com/JuliaTesting/TestEnv.jl) for this purpose.

**Comment by jishnub** [comment]
From what I understand, it is simply ignored on earlier versions, so it shouldn't be an issue, I guess?

**Comment by johnnychen94** [comment]
Never ever tried both, though.

But if you think `test/Project.toml` is more convenient than TestEnv, then I'm okay with that.

**Comment by jishnub** [comment]
I think it's simpler, and test dependencies aren't updated frequently anyway for it to matter much.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1 @@
-/Manifest.toml
-/docs/Manifest.toml
+*Manifest.toml
```
Comment (johnnychen94) [inline .gitignore:1]:
```suggestion
Manifest.toml
```

**Review by johnnychen94 — APPROVED** [review]

**Pull request merged**